doc/examples/compactor/compactor_actors_example.py

```diff
@@ -0,0 +1,332 @@
+    hasher = hashlib.sha1()
+    hasher.update(pk_bytes)
+    return hasher.digest()
+
+
+#########################
+# Event Timestamp Utils #
+#########################
+def read_files_add_event_timestamp(file_paths, event_timestamp_column_name):
+
+    input_file_to_df = read_parquet_files(file_paths)
+    for input_file, dataframe in input_file_to_df.items():
+        event_timestamp = get_input_file_event_timestamp(input_file)
+        dataframe[event_timestamp_column_name] = int(event_timestamp)
+    return concat_dataframes(input_file_to_df.values())
+
+
+def get_input_file_event_timestamp(input_file):
+    input_file_name = path.basename(input_file)
+    m = re.match(".+?-(?P<_0>\d+)\.[parq|parquet]", input_file_name)
+    return next(iter(m.groupdict().values()))
+
+
+#################
+# Preconditions #
+#################
+def check_preconditions(primary_keys, sort_keys, max_records_per_output_file):
+
+    assert len(primary_keys) == len(set(primary_keys)), \
+        "Primary key names must be unique: {}".format(primary_keys)
+    assert len(sort_keys) == len(set(sort_keys)), \
+        "Sort key names must be unique: {}".format(sort_keys)
+    assert max_records_per_output_file >= 1, \
+        "Max records per output file must be a positive value"
+
+
+#############
+# Compactor #
+#############
+def compact(bucket, table_stream_id, primary_keys, sort_keys,
+            max_records_per_output_file, num_hash_buckets):
+
+    # check preconditions before doing any computationally expensive work
+    check_preconditions(
+        primary_keys,
+        sort_keys,
+        max_records_per_output_file,
+    )
+
+    # define distinct, but constant, pk hash and event timestamp column names
+    col_uuid = "4000f124-dfbd-48c6-885b-7b22621a6d41"
+    pk_hash_column_name = "{}_hash".format(col_uuid)
+    event_timestamp_column_name = "{}_event_timestamp".format(col_uuid)
+
+    # append the event timestamp column to the sort key list
+    sort_keys.append(event_timestamp_column_name)
+
+    # discover input file paths
+    input_file_paths = filter_file_paths_by_prefix(
+        bucket, "input/{}".format(table_stream_id))
+    all_hash_bucket_indices = set()
+
+    # create an actor to track dataset chunks for each hash bucket
+    hb_actors = []
+    for i in range(num_hash_buckets):
+        hb_actors.append(HashBucket.remote())
+
+    # group like primary keys together by hashing them into buckets
+    hb_tasks_pending = []
+    for input_file_path in input_file_paths:
+        hb_task_promise = hash_bucket.remote(
+            [input_file_path],
+            primary_keys,
+            num_hash_buckets,
+            pk_hash_column_name,
+            event_timestamp_column_name,
+            hb_actors,
+        )
+        hb_tasks_pending.append(hb_task_promise)
+    while len(hb_tasks_pending):
+        hb_task_complete, hb_tasks_pending = ray.wait(hb_tasks_pending)
+        all_hash_bucket_indices.update(ray.get(hb_task_complete[0]))
```

Comment:
In this case, ray.wait(hb_tasks_pending) implicitly sets the default value of num_returns to 1, so it should only ever return 1 item at a time in hb_task_complete.

However, since we're not doing anything useful here by waiting for one item at a time, I suppose we could get rid of the wait and while loop altogether to just write:

``` python
    hb_tasks_complete = ray.get(hb_tasks_pending)
    all_hash_bucket_indices.update(itertools.chain.from_iterable(hb_tasks_complete))
```